File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        differenceMap={}
        
        for i in range(0,len(nums)):
            num=nums[i]
            if num in differenceMap.keys():
                return [differenceMap[num][0],i]
            else:
                difference=target-num
                if difference in differenceMap.keys():
                    differenceMap[difference].append(i)
                else:
                    differenceMap[difference]=[i]
        

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {

        int first=0;
        int last=numbers.length-1;
        while(last>first)
        {
            if((numbers[first]+numbers[last])>target){
                last--;
            }
            else if ((numbers[first]+numbers[last])<target)
            {
                first++;
            }
            else if((numbers[first]+numbers[last])==target)
            {
                return new int[]{first+1,last+1};
            }
        }
        return new int[]{-1,-1};
        
    }
}
